lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1210,48 +1212,47 @@ def main():
         module.fail_json(msg='. '.join(error_msgs))
 
     params = module.params
+    package_list = [ package.strip() for package in params['name']]
+    exclude = params['exclude']
+    state = params['state']
+    enablerepo = params.get('enablerepo', '')
+    disablerepo = params.get('disablerepo', '')
+    disable_gpg_check = params['disable_gpg_check']
+    skip_broken = params['skip_broken']
+    conf_file = params['conf_file']
 
     if params['list']:
-        repoquerybin = ensure_yum_utils(module)
+        repoquerybin = module.get_bin_path('repoquery', required=False)
         if not repoquerybin:
             module.fail_json(msg="repoquery is required to use list= with this module. Please install the yum-utils package.")
         results = dict(results=list_stuff(module, repoquerybin, params['conf_file'], params['list'], params['installroot']))
+        module.exit_json(**results)
+
+    """ If rhn-plugin is installed and no rhn-certificate is available on
+    the system then users will see an error message using the yum API.
+    Use repoquery in those cases. """
 
+    my = yum_base(params['conf_file'], params['installroot'])
+    my.conf # A sideeffect of accessing conf is that the configuration is loaded and plugins are discovered, this is used to veryif if rhnplugin is in enabled
+    repoquery = None
+    try:
+        yum_plugins = my.plugins._plugins
+    except AttributeError:
+        pass
     else:
-        # If rhn-plugin is installed and no rhn-certificate is available on
```

Comment:
broke this else due to 2 reasons. 

1) plugin has to exit right after `list` completion as per module documentation that states that list and name are mutually exclusive. 
2) ansible will not allow list to be executed with name parameters in the first place using mutual exclusive mechanism.